cli/lockfile.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use std::path::PathBuf;
+
+use deno_core::anyhow::Context;
+use deno_core::error::AnyError;
+use deno_core::parking_lot::Mutex;
+use deno_core::parking_lot::MutexGuard;
+use deno_core::ModuleSpecifier;
+use deno_runtime::deno_node::PackageJson;
+
+use crate::args::ConfigFile;
+use crate::cache;
+use crate::util::fs::atomic_write_file_with_retries;
+use crate::Flags;
+
+use crate::args::DenoSubcommand;
+use crate::args::InstallFlags;
+use crate::args::InstallKind;
+
+use deno_lockfile::Lockfile;
+
+#[derive(Debug)]
+pub struct CliLockfile {
+  lockfile: Mutex<Lockfile>,
+}
+
+pub struct Guard<'a, T> {
+  guard: MutexGuard<'a, T>,
+}
+
+impl<'a, T> std::ops::Deref for Guard<'a, T> {
+  type Target = T;
+
+  fn deref(&self) -> &Self::Target {
+    &self.guard
+  }
+}
+
+impl<'a, T> std::ops::DerefMut for Guard<'a, T> {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    &mut self.guard
+  }
+}
+
+impl CliLockfile {
+  pub fn new(lockfile: Lockfile) -> Self {
+    Self {
+      lockfile: Mutex::new(lockfile),
+    }
+  }
+
+  pub fn specifier(&self) -> Option<ModuleSpecifier> {
+    let path = &self.lockfile.lock().filename;
+    if path.exists() {
+      Some(ModuleSpecifier::from_file_path(path).ok()?)
+    } else {
+      None
+    }
+  }
+
+  /// Get the inner deno_lockfile::Lockfile.
+  pub fn inner(&self) -> Guard<Lockfile> {
```

Comment:
Maybe let's call this `lock()` or `inner_lock()` just to be clear in the code above that it's locking?

Alternatively, maybe we could do something like:

```rs
lockfile.with_inner_mut(|lockfile| {
  // in here you can modify or return information
});
```

Reply:
Just went with `lock()`, I'm not a huge fan of the indentation make it a closure would add.